Document/0x07a-Testing-Authentication-and-Session-Management.md

```diff
@@ -8,6 +8,65 @@ For all of the test cases below, it need to be investigated first what kind of a
 
 Cookie-Based Authentication is the traditional authentication mechanism used in web applications. In order to adopt to the different requirements of mobile apps Token-Based Authentication was specified and is more and more used nowadays in mobile apps. A prominent example for this is JSON Web Token or JWT<sup>[1]</sup> which can be part of an OAuth2 authentication and authorization framework.
 
+#### OAuth2
+
+OAuth2 is an authorization framework used to authorize an application to use a user account on an HTTP service for a limited time and, at the same time, preventing the client applications from having knowledge of any user credentials.
+
+OAuth2 defines 4 roles:
+
+* Resource Owner: the user owning the account.
+* Client: the application that wants to access the user's account using the access tokens.
+* Resource Server: hosts the user accounts.
+* Authorization Server: verifies the identity of the user and issues access tokens to the application.
+
+Note: the API fulfills both the resource and authorization server roles. Therefore we will refer to both as the API.
+
+<img src="Images/Chapters/0x07a/abstract-oauth2-flow.png" width="350px"/>
+
+Here is a more detailed explanation of the steps in the diagram:
+
+1. The application requests authorization to access service resources from the user
+2. If the user authorized the request, the application receives an authorization grant
+3. The application requests an access token from the authorization server (API) by presenting authentication of its own identity, and the authorization grant
+4. If the application identity is authenticated and the authorization grant is valid, the authorization server (API) issues an access token to the application. Authorization is complete.
+5. The application requests the resource from the resource server (API) and presents the access token for authentication
+6. If the access token is valid, the resource server (API) serves the resource to the application
+
+Source: https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2
+
+The authorization grant might have different forms (explicit, implicit, etc), the access token might have a companion refresh token and might be used on different ways (e.g., as a bearer token). Please refer to https://tools.ietf.org/html/rfc6749 for further details.
```

Comment:
Would it be a good idea to put this sentence within the flow-steps itself? As this is often misinterpreted ....